src/test/groovy/org/prebid/server/functional/tests/CookieSyncSpec.groovy

```diff
@@ -0,0 +1,516 @@
+package org.prebid.server.functional.tests
+
+import org.prebid.server.functional.model.UidsCookie
+import org.prebid.server.functional.model.config.AccountCcpaConfig
+import org.prebid.server.functional.model.config.AccountConfig
+import org.prebid.server.functional.model.config.AccountPrivacyConfig
+import org.prebid.server.functional.model.db.Account
+import org.prebid.server.functional.model.request.cookiesync.CookieSyncRequest
+import org.prebid.server.functional.model.request.cookiesync.FilterSettings
+import org.prebid.server.functional.model.response.cookiesync.CookieSyncResponse
+import org.prebid.server.functional.service.PrebidServerService
+import org.prebid.server.functional.util.HttpUtil
+import org.prebid.server.functional.util.PBSUtils
+import org.prebid.server.functional.util.privacy.CcpaConsent
+import org.prebid.server.functional.util.privacy.TcfConsent
+
+import java.time.Instant
+
+import static org.prebid.server.functional.model.bidder.BidderName.ALIAS
+import static org.prebid.server.functional.model.bidder.BidderName.BOGUS
+import static org.prebid.server.functional.model.bidder.BidderName.GENERIC
+import static org.prebid.server.functional.model.bidder.BidderName.RUBICON
+import static org.prebid.server.functional.testcontainers.Dependencies.getNetworkServiceContainer
+import static org.prebid.server.functional.util.privacy.CcpaConsent.Signal.ENFORCED
+import static org.prebid.server.functional.util.privacy.TcfConsent.GENERIC_VENDOR_ID
+import static org.prebid.server.functional.util.privacy.TcfConsent.PurposeId.BASIC_ADS
+
+class CookieSyncSpec extends BaseSpec {
+
+    PrebidServerService prebidServerService = pbsServiceFactory.getService(
+            ["adapters.generic.usersync.redirect.url"            : "$networkServiceContainer.rootUri/generic-usersync&redir={{redirect_url}}".toString(),
+             "adapters.generic.usersync.redirect.support-cors"   : "false",
+             "adapters.generic.usersync.redirect.format-override": "blank"])
+
+    def "PBS cookie sync with host-cookie.family which hasn't the same value in both uids and from the direct host-cookie then should re-sync"() {
+        given: "PBS bidders config"
+        def prebidServerService = pbsServiceFactory.getService(
+                ["host-cookie.family"                                : "generic",
```

Comment:
This name should correspond to bidder name. Please or should correlate with family name override. Please make this obvious.